api/envoy/config/filter/udp/dns_filter/v2alpha/dns_filter.proto

```diff
@@ -0,0 +1,86 @@
+syntax = "proto3";
+
+package envoy.config.filter.udp.dns_filter.v2alpha;
+
+import "envoy/api/v2/core/base.proto";
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+
+import "udpa/annotations/migrate.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.filter.udp.dns_filter.v2alpha";
+option java_outer_classname = "DnsFilterProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package =
+    "envoy.extensions.filter.udp.dns_filter.v3";
+option (udpa.annotations.file_status).work_in_progress = true;
+
+// [#protodoc-title: DNS Filter]
+// DNS Filter :ref:`configuration overview <config_udp_listener_filters_dns_filter>`.
+// [#extension: envoy.filters.udp_listener.dns_filter]
+
+// Configuration for the DNS filter.
+message DnsFilterConfig {
```

Comment:
I think this probably should live in `envoy.data`. This is basically the DNS table, so it's a data format, rather than filter configuration. It can come via either a `DataSource` or inline.

Reply:
I see your point.  Let me see what I can do with this.